### ISSUE TITLE
Allow inheritance of `.bazelrc` snippets/flags across projects

### ISSUE THREAD

**Issue by armandomontanez** [bazelbuild/bazel#23028]
### Description of the feature request:

One of the side-effects of incompatible Bazel flags in the Bazel ecosystem is that some projects *require* certain incompatible flags to be set in order for the project to build at all. As middleware, managing these flags can be particularly tricky since you may require different incompatible flags depending on the version of your dependencies, and you also may prescribe your own required flags. These are forked in downstream users projects, and they may accidentally diverge in ways that cause subtle or confusing breakages as updates are rolled out. Because there's no way to `import` .bazelrc snippets from external dependencies, there's virtually no solution for inheritance at all and you just have to manually manage your own project's .bazelrc and hope for the best.

Perhaps the right way to handle this is to integrate it into bzlmod, as that mechanism strives to address similar problems that existed in WORKSPACE files. Admittedly, this would likely be a massive undertaking to design and implement.

### Which category does this issue belong to?

External Dependency

### What underlying problem are you trying to solve with this feature?

Inconsistencies in `.bazelrc` requirements across dependent projects.

**Comment by Wyverald** [comment]
cc @gregestren who's working on related issues.

**Comment by fmeum** [comment]
> Rolling out the change that required that involved updating various downstream projects to include that flag as well. 

Could you explain what you mean by that? The `.bazelrc` files of external repos aren't used by Bazel, so flipping the flag in Pigweed's own `.bazelrc` should be sufficient.

**Comment by armandomontanez** [comment]
> The .bazelrc files of external repos aren't used by Bazel, so flipping the flag in Pigweed's own .bazelrc should be sufficient.

Right, it's sufficient for *Pigweed* to build, but then making said incompatible change with the associated flag then breaks downstream projects, which is the entire problem. When I roll downstream users, I have to make sure that when I update their reference to Pigweed I also update their `.bazelrc` with new required flags. If I could instead have a list of "flags required to successfully build pigweed," then whenever I roll Pigweed in downstream projects the'll implicitly pick up the new list.

 Here's a step-by-step example:
1. `@pigweed//:libfoo` has a known issue
2. `@pigweed//:libfoo` is fixed, but requires setting `--incompatible_default_to_explicit_init_py` Bazel flag in Pigweed's `.bazelrc`.
3. Someone tries to build downstream project `@myproj` against the new version of Pigweed, and it breaks.
4. Downstream project `@myproj`  updates their `.bazelrc` to include `--incompatible_default_to_explicit_init_py` as part of the update to `@pigweed`, and their build succeeds again.

This required manual intervention isn't just inconvenient; sometimes the breakages are extremely subtle and don't manifest until long after the update. We end up with a lot of fragmentation across various `.bazelrc` files since there's no way to to scalably and maintainably centralize these critical required flags as they evolve over time.

**Comment by fmeum** [comment]
Including a `.bazelrc` snippet pro used by pigweed could have the reverse problem though: By rolling forward pigweed, the downstream project could break if it's incompatible with the new flag.

Could you add a `config_setting` matching the required flag value and then add a `select` on it with a descriptive `no_match_error` to your core targets? That way users would see the error if the flag values mismatch, but would still need to take explicit action (which avoids spooky action at a distance).

**Comment by gregestren** [comment]
I appreciate the basic issue.

I also don't know what the right resolution is. 

Knowing if a flag is safe has to be a combined assessment of Pigweed and the downstream project, right? Each for their own reasons.

If we could more easily model these requirements (which we can certainly think about), what's the right way for Bazel to communicate to the user how the cumulative constraints resolve? And how automatic do we want flag flips to be, without users explicitly reviewing / setting them?

**Comment by armandomontanez** [comment]
This is still a relevant (but very difficult to address) problem.
